src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,42 @@
+package duke.parser;
+
+import duke.command.Command;
+import duke.command.DoneCommand;
+import duke.command.DeleteCommand;
+import duke.command.FindCommand;
+import duke.command.ExitCommand;
+import duke.command.ListCommand;
+import duke.command.AddCommand;
+import duke.exception.DukeException;
+
+public class Parser {
+
+    /**
+     * Parse the command entered by user into a Command object if command exists.
+     *
+     * @param  command  the input by the user
+     * @return  the parsed command
+     * @throws  DukeException   if command doesn't exist
+     */
+    public static Command parse(String command) throws DukeException {
+        String[] commandTokens = command.split(" ");
+        switch (commandTokens[0]) {
```

Comment:
beware that if the input string contains only space, then commandTokens[0] will be null, and it will trigger index out of bound exception. same for all other uses of this split(" ") in your code.